packages/request-manager/src/events/bootstrap.ts

```diff
@@ -0,0 +1,27 @@
+// Section 4.1.10. Status events in https://gitweb.torproject.org/torspec.git/tree/control-spec.txt
+
+import { BootstrapEvent } from '../types';
+
+const clientStatusGroupParser = (message: string): RegExpMatchArray | null =>
+    message.trim().match(/^(^650 STATUS_CLIENT NOTICE BOOTSTRAP PROGRESS=.*)+/gm);
+
+const statusParser = (status: string): RegExpMatchArray | null =>
+    status.trim().match(/^650 STATUS_CLIENT NOTICE BOOTSTRAP (PROGRESS=(?<progress>\d+)?.*)?/);
+
+export const bootstrapParser = (message: string): BootstrapEvent[] => {
```

Comment:
I didn't like the double-regex-testing, so I've tried to simplify the function and came to this:
```
export const bootstrapParser = (message: string): BootstrapEvent[] =>
    [...message.matchAll(/^\s*650 STATUS_CLIENT NOTICE BOOTSTRAP PROGRESS=(?<progress>\d+)/gm)].map(
        match => ({ progress: match.groups?.progress ?? '' }),
    );
```
a) I'm not sure it is equivalent, but hopefully it is.
b) Unfortunately, TS configuration must be changed in order to use `matchAll` (specifically, `lib: ["es2020"] `) and I don't know if we should do it.
c) We may want to add other events as well in the future, and then this could be not the best approach.

In the end, I'm not so sure about it, so you could leave it as it is or implement some middle way (my regex, but iterate without `matchAll`), etc. Your call.

Reply:
I will think about it.